python/dgl/graphbolt/impl/fused_csc_sampling_graph.py

```diff
@@ -923,3 +1040,27 @@ def from_dglgraph(
         ),
         metadata,
     )
+
+
+def _get_nodetype(index, node_type_offset):
+    """Get node type through node_type_offset."""
+    ntype = -1
+    for offset in node_type_offset:
+        if index >= offset:
+            ntype += 1
+        else:
+            break
+    return ntype
+
+
+def _binary_search_etype_end(type_per_edge, l, r, etype_id):
```

Comment:
See https://pytorch.org/docs/stable/generated/torch.searchsorted.html. but I'm not sure it 100% applied to this case.